lib/reek/source/source_code.rb

```diff
@@ -39,10 +40,10 @@ def initialize(code, origin, parser: Parser::Ruby22)
       # :reek:DuplicateMethodCall: { max_calls: 2 }
       def self.from(source)
         case source
-        when File     then new(source.read, source.path)
-        when IO       then new(source.readlines.join, IO_IDENTIFIER)
-        when Pathname then new(source.read, source.to_s)
-        when String   then new(source, STRING_IDENTIFIER)
+        when File       then new(source.read, source.path)
+        when IO         then new(source.readlines.join, IO_IDENTIFIER)
+        when SourcePath then new(source.read, source.to_s)
```

Comment:
Examiner should still take a Pathname. Otherwise, we break the API. Also, I don't want to force people to have to wrap their Pathname in a SourcePath.